lib/Page.js

```diff
@@ -495,7 +495,12 @@ class Page extends EventEmitter {
   async _onConsoleAPI(event) {
     const context = this._frameManager.executionContextById(event.executionContextId);
     const values = event.args.map(arg => createJSHandle(context, arg));
-    this._addConsoleMessage(event.type, values);
+    let location;
+    if (event.stackTrace && event.stackTrace.callFrames) {
+      const {url, lineNumber, columnNumber} = event.stackTrace.callFrames[0];
+      location = {url, lineNumber, columnNumber};
```

Comment:
just a clarification about how defensive puppeteer code should be:

Can i rely on devtools protocol API for optional/mandatory return values?
e.g.
* [Runtime#type-CallFrame](https://chromedevtools.github.io/devtools-protocol/tot/Runtime#type-CallFrame) always return `{url, lineNumber, columnNumber}`.
* [Runtime#event-consoleAPICalled](https://chromedevtools.github.io/devtools-protocol/tot/Runtime#event-consoleAPICalled) might not return `StackTrace`. But if returned, it will always return `Array{CallFrame}`.

Reply:
done